src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -1485,6 +1485,8 @@ static QVariant fcnRegexpReplace( const QVariantList &values, const QgsExpressio
   QString str = QgsExpressionUtils::getStringValue( values.at( 0 ), parent );
   QString regexp = QgsExpressionUtils::getStringValue( values.at( 1 ), parent );
   QString after = QgsExpressionUtils::getStringValue( values.at( 2 ), parent );
+  if ( after.isNull() )
+    after = QString( "" );
```

Comment:
Is this change actually required? I think the change to the function definition alone should be sufficient to fix the issue. Null QStrings are mostly a historical oddity...

Reply:
Nope sorry, old and bad habit